x/subspaces/ante/tx_checker.go

```diff
@@ -0,0 +1,48 @@
+package ante
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/ante"
+	antetypes "github.com/desmos-labs/desmos/v5/x/subspaces/ante/types"
+)
+
+// CheckTxFeeWithSubspaceMinPrices returns the tx checker that including the subspace allowed tokens into minimum prices list
+func CheckTxFeeWithSubspaceMinPrices(txFeeChecker ante.TxFeeChecker, sk antetypes.SubspacesKeeper) ante.TxFeeChecker {
+	return func(ctx sdk.Context, tx sdk.Tx) (sdk.Coins, int64, error) {
+		subspaceID, isSubspaceTx := GetTxSubspaceID(tx)
+		if !isSubspaceTx {
+			return txFeeChecker(ctx, tx)
+		}
+
+		subspace, found := sk.GetSubspace(ctx, subspaceID)
+		if !found {
+			return txFeeChecker(ctx, tx)
+		}
+
+		newMinPrices := MergeMinPrices(ctx.MinGasPrices(), sdk.NewDecCoinsFromCoins(subspace.AllowedFeeTokens...))
+		newCtx := ctx.WithMinGasPrices(newMinPrices)
+		return txFeeChecker(newCtx, tx)
```

Comment:
I'm not super familiar with on-chain gas prices, but I thought they were treated using the `and` logic. So if I have a min gas price of `0.01udsm,0.01ustar` then to properly broadcast my transaction I have to pay fees in *both* tokens, and not *either one* of the two. Do you have any link for this to be checked out that we can paste here to remember this in the future?

Reply:
They are treated as `or`,  so one of tokens reaches the min fees can broadcast properly, say, to send the message with 100 gas via the node with `0.01udsm, 0.01ustar` min gas prices, `1udsm` or `1ustar` can pass the check.

Here is the reference:
https://github.com/cosmos/cosmos-sdk/blob/release/v0.47.x/x/auth/ante/validator_tx_fee.go#L38